src/algorithms/AsymmetricForwardBackwardAdjoint.jl

```diff
@@ -0,0 +1,197 @@
+################################################################################
+# Primal-dual algorithms based on Asymmetric Forward-Backward-Adjoint 
+#
+# The iterator implements Algorithm 3 of [1] with constant stepsize (α_n=λ) for several prominant special cases.
+# [1] Latafat, Patrinos. "Asymmetric forward–backward–adjoint splitting for solving monotone inclusions involving three operators"  Computational Optimization and Applications, pages 1–37, 2017.
+
+
+struct AFBAIterator{I <: Integer, R <: Real, T <: BlockArray} <: ProximalAlgorithm{I, T}
+    x::T
+    y::T
+    g
+    h
+    f
+    l
+    L
+    theta::R
+    mu::R
+    lam::R
+    betaQ::R # Lipschitz constant of f
+    betaR::R # Lipschitz constant of l
+    gamma1::R
+    gamma2::R
+    maxit::I
+    tol::R
+    verbose::I
+    verbose_freq::I
+    xbar::T
+    ybar::T
+    gradf::T
+    gradl::T
+    FPR_x::T
+    FPR_y::T
+end
+
+################################################################################
+# Constructor(s)
+
+function AFBAIterator(x0::T, y0::T; g=Zero(), h=Zero(), f=Zero(), l=Zero(), L=Identity(blocksize(x0)), theta=2, mu=1, lam=1, betaQ=0, betaR=0, gamma1=-1, gamma2=-1, maxit::I=10000, tol::R=1e-5, verbose=1, verbose_freq = 100) where {I, R, T}
```

Comment:
Two more things: I notice that the names of the terms are swapped wrt the AFBA paper mentioned above. This is fine, but then make sure to include a docstring for the `AFBA` function where you say "the problem is this and has these terms, the assumptions are these, the other arguments are this and that, and bla bla".

Concerning `l`, I guess it would be better to provide the function itself rather than the conjugate, so as to avoid confusion: in that case `l` should default to `Conjugate(ProximalOperators.Zero())`, correct? Then internally you conjugate again (I'm fixing ProximalOperators to make sure `Conjugate(Conjugate(f)) == f` in case `f` is convex, currently this is not the case: it still works correctly but I guess it is a bit inefficient).

Reply:
Or even better: you can use directly `ProximalOperators.IndZero`! Definitely go for that.